x-pack/filebeat/module/sophos/xg/ingest/pipeline.yml

```diff
@@ -28,6 +28,13 @@ processors:
 - set:
     field: _temp_.time
     value: "{{sophos.xg.date}} {{sophos.xg.time}}"
+    if: "ctx?.sophos?.xg?.date != null && ctx?.sophos?.xg?.time != null"
+
+- set:
+    field: _temp_.time
+    value: "{{sophos.xg.timestamp}}"
+    if: "ctx?.sophos?.xg?.timestamp != null"
+
 - date:
     if: "ctx?.event?.timezone == null"
     field: _temp_.time
```

Comment:
As the two set processors above are now conditional, technically these two date processors could execute with `_temp_.time` being unset. I suggest extending the condition for date so that they are not run when `_temp_.time` is not set, or add an `ignore_failure: true`.